src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -118,59 +118,70 @@ public DefaultSerializer(final ILogger logger) {
      * @param serializedObject   the parent object whose children will be iterated to set additional data
      * @param rawJson            the raw json
      */
-    @SuppressWarnings("unchecked")
-    private void setChildAdditionalData(IJsonBackedObject serializedObject, JsonObject rawJson) {
+    private void setChildAdditionalData(final IJsonBackedObject serializedObject, final JsonObject rawJson) {
         // Use reflection to iterate through fields for eligible Graph children
-        for (java.lang.reflect.Field field : serializedObject.getClass().getFields()) {
-            try {
-                Object fieldObject = field.get(serializedObject);
-
-                // If the object is a HashMap, iterate through its children
-                if (fieldObject instanceof HashMap) {
-                    HashMap<String, Object> serializableChildren = (HashMap<String, Object>) fieldObject;
-                    Iterator<Entry<String, Object>> it = serializableChildren.entrySet().iterator();
+        if(rawJson != null) {
+            for (java.lang.reflect.Field field : serializedObject.getClass().getFields()) {
+                try {
+                    if(field != null) {
+                        final Object fieldObject = field.get(serializedObject);
+                        if (fieldObject instanceof HashMap) {
```

Comment:
Add a test for `fieldObject instanceof HashMap` using `serializer.deserializeObject(sourcestring, class);`